### ISSUE TITLE
Resize gradients computed by basic math Functions to match original input sizes

### ISSUE THREAD

**Issue by Kontrakt** [pytorch/pytorch#610]
**Code to reproduce:**

```python
import torch
from torch.autograd import Variable

a = Variable(torch.randn(10))
b = Variable(torch.randn(10, 5), requires_grad=True)
c = torch.norm(b, 2, 1)
d = torch.div(a, c)
loss = torch.sum(d)
loss.backward()
```

**The problem:**

torch.norm  produces variable with shape `[10, 1]`  and div returns `[10]`, this breaks backward because sizes must be equal.

**Comment by Kontrakt** [comment]
@fmassa You are right, it's not hard to bypass it, but it feels a bit annoying to get such errors.

In torch `gradInput` was resized to the input shapes [example](https://github.com/torch/nn/blob/e1efc6345f3dec8b631f91f640c11a4b7dd9e012/CMul.lua#L80). Are there reasons we can't do the same in pytorch?

**Comment by apaszke** [comment]
Yeah, we should fix that. I forgot that the sizes don't have to match in these ops.

**Comment by soumith** [comment]
this was fixed at some point, we forgot to close the issue.
